source/code/omi_preexec/omi-mysql.cpp

```diff
@@ -82,113 +83,19 @@ static int MySQL_ConfigurationParser(uid_t uid, gid_t gid, const std::wstring& c
         return 0;
     }
 
-    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Parsing MySQL configuration file: ", cnfFile));
+    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Loading MySQL configuration file: ", cnfFile));
 
-    std::vector<std::wstring> lines;
-    try
+    MySQL::MySQL_ConfigurationScanner configScanner;
+    if (!configScanner.Scan(cnfFile, auth))
     {
-        SCXStream::NLFs nlfs;
-        SCXFile::ReadAllLines(cnfFile, lines, nlfs );
-    }
-    catch (const SCXException& e)
-    {
-        SCX_LOGERROR(hLog, StrAppend(L"Error reading configuration file: ", e.What()));
+        SCX_LOGERROR(hLog, StrAppend(L"Error reading ", cnfFile));
```

Comment:
Did you get to this? I didn't see that ...

---
In reply to: [66894954](https://github.com/Microsoft/MySQL-Provider/pull/5#discussion_r66894954) [](ancestors = 66894954)